skyportal/models/observation_plan.py

```diff
@@ -0,0 +1,347 @@
+        doc='End of validity window',
+    )
+
+    status = sa.Column(
+        sa.String(),
+        nullable=False,
+        default="pending submission",
+        index=True,
+        doc="The status of the observing plan.",
+    )
+
+    planned_observations = relationship(
+        "PlannedObservation",
+        passive_deletes=True,
+        doc='Planned observations associated with this plan.',
+    )
+
+    @property
+    def start_observation(self):
+        """Time of the first planned observation."""
+        if self.planned_observations:
+            return min(
+                [
+                    planned_observation.obstime
+                    for planned_observation in self.planned_observations
+                ]
+            )
+        else:
+            return None
+
+    @hybrid_property
+    def num_observations(self):
+        """Number of planned observation."""
+        return len(self.planned_observations)
+
+    @num_observations.expression
+    def num_observations(cls):
+        """Number of planned observation."""
+        return cls.planned_observations.count()
+
+    @property
+    def total_time(self):
+        """Total observation time (seconds)."""
+        return sum(_.exposure_time for _ in self.planned_observations)
+
+    @property
+    def tot_time_with_overheads(self):
+        overhead = sum(_.overhead_per_exposure for _ in self.planned_observations)
+        return overhead + self.total_time
+
+    @property
+    def ipix(self):
+        return {
+            i
+            for planned_observation in self.planned_observations
+            if planned_observation.field.ipix is not None
+            for i in planned_observation.field.ipix
+        }
+
+    @property
+    def area(self):
+        nside = Localization.nside
+        return hp.nside2pixarea(nside, degrees=True) * len(self.ipix)
+
+    def get_probability(self, localization):
+        ipix = np.asarray(list(self.ipix))
+        return localization.flat_2d[ipix].sum()
+
+
+class PlannedObservation(Base):
+    """A single planned exposure as part of an EventObservationPlan.
+    Tile information, including the event time, localization ID, field IDs,
+    tiling name, and tile probabilities."""
+
+    observation_plan_id = sa.Column(
+        sa.ForeignKey('eventobservationplans.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Event observation plan ID',
+    )
+
+    observation_plan = relationship(
+        "EventObservationPlan",
+        foreign_keys=observation_plan_id,
+        doc="The EventObservationPlan that this planned observation belongs to",
+    )
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    dateobs = sa.Column(
+        sa.ForeignKey('gcnevents.dateobs', ondelete="CASCADE"),
+        nullable=False,
+        index=True,
+        doc='UTC event timestamp',
```

Comment:
Same as above. In particular when you have both `dateobs` and `obstime` that have very different meanings.

Reply:
same here.